go/arrow/array/fixed_size_list.go

```diff
@@ -110,6 +112,44 @@ func (a *FixedSizeList) Release() {
 	a.values.Release()
 }
 
+func (a *FixedSizeList) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+	slice := a.newListValue(i)
+	defer slice.Release()
+	v, err := json.Marshal(slice)
+	if err != nil {
+		panic(err)
+	}
+
+	return json.RawMessage(v)
+}
+
+func (a *FixedSizeList) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+
+	buf.WriteByte('[')
```

Comment:
Is this using a string to encode FixedSizeLists?  That might be counter-intuitive to people?  I guess this is needed to round-trip with full-fidelity?

Reply:
No, it's just encoding a fixed size list identically to how it encodes regular lists, as a nested list. From the perspective of JSON there's no difference between a `List` where all of the lists happen to be the same length, and a `FixedSizeList`. A fixed size list with `n=3` of Int would encode in json as `[ [1,2,3], [4,5,6], [7,8,9] ]`

Reply:
I can understand the confusion now that I think about it. The return from `MarshalJSON` is the byte slice of the raw JSON that is produced by the conversion. I hope that clears things up a bit.